### PR TITLE
Send confirmation email once ticket has been assigned

### PR DESCRIPTION
**Pull request by vidya-ram**
1. Send confirmation email to assignee and to buyer, once user fills in their details and ticket has been assigned. Also when ticket has been transferred.
2. Change enqueue to use @job.
3. Remove slack web hook.

1. Order confirmation on the website:
screen shot 2016-08-09 at 6 14 08 pm

2. Ticket confirmation mail:
screen shot 2016-08-10 at 5 44 19 pm

### CODE REVIEW

File: boxoffice/templates/ticket_assigment.html [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    }
+  </style>
+</head>
+
+<body>
+  <div>
+
+    <p>Hello {{ line_item.current_assignee.fullname }},</p>
+
+    <p>Thank you for filling in the details. {{line_item.item.title}} has been assigned to you.</p>
```
Comment (shreyas-satish) [inline boxoffice/templates/ticket_assigment.html]: Do we need 'Thank you for filling in the details.'. One possibility is, the assignee may not have filled the details themselves.

File: boxoffice/views/participant.py [hunk 2]
```diff
@@ -35,4 +41,5 @@ def assign(order):
         phone=assignee_dict.get('phone'), details=assignee_details, line_item=line_item)
         db.session.add(assignee)
     db.session.commit()
+    boxofficeq.enqueue(send_ticket_assignment_email, order.id, line_item.id)
```
Comment (shreyas-satish) [inline boxoffice/views/participant.py]: This should only happen when a new assignee is created, right?

Comment (shreyas-satish) [inline boxoffice/views/participant.py]: Also, you can just pass `line_item.id`. You can derive `order` with `line_item.order`.

File: boxoffice/views/participant.py [hunk 3]
```diff
@@ -34,5 +40,6 @@ def assign(order):
         assignee = Assignee(current=True, email=assignee_dict.get('email'), fullname=assignee_dict.get('fullname'),
         phone=assignee_dict.get('phone'), details=assignee_details, line_item=line_item)
         db.session.add(assignee)
+        boxofficeq.enqueue(send_ticket_assignment_email, line_item.id)
```
Comment (jace) [inline boxoffice/views/participant.py]: Don't use `enqueue`. Add an `@job('boxoffice')` decorator to the function and use `send_ticket_assignment_email.delay(...)` here.

**Comment by jace** [comment]
@vidya-ram in general, asking people to click on a button that's right next to the caption is redundant. Best to remove that sentence.

Also, since getting the ticket is higher priority than getting the receipt, reorder the buttons.

File: boxoffice/mailclient.py [hunk 4]
```diff
@@ -2,27 +2,33 @@
 
 from decimal import Decimal
 from flask import render_template
+from rq.decorators import job
```
Comment (jace) [inline boxoffice/mailclient.py]: This should be from `from flask.ext.rq import job`. [See documentation](http://flask-rq.readthedocs.io/en/latest/).

File: boxoffice/mailclient.py [hunk 5]
```diff
@@ -2,27 +2,33 @@
...
+from redis import Redis
 from flask.ext.mail import Message
 from html2text import html2text
 from premailer import transform as email_transform
 from .models import Order, LineItem, LINE_ITEM_STATUS, CURRENCY_SYMBOL
 from . import mail, app
 
+redis_connection = Redis()
```
Comment (jace) [inline boxoffice/mailclient.py]: This is not required.

File: boxoffice/mailclient.py [hunk 6]
```diff
@@ -2,27 +2,33 @@
...
 
-def send_receipt_email(order_id, subject="Thank you for your order!"):
+
+@job('boxoffice', connection=redis_connection)
```
Comment (jace) [inline boxoffice/mailclient.py]: If you use Flask-RQ's `@job`, you don't need to specify the connection.

File: boxoffice/templates/boxoffice.html [hunk 7]
```diff
@@ -166,11 +166,10 @@ <h1 class="category-heading">{{title}}</h1>
           <p class="confirmation-icon"><i class="fa fa-check-circle"></i></p>
           <p class="confirmation-header">Thank you for your order!</p>
           <div class="confirmation-msg">
-            <p class="name">Hello {{buyer.name}},</p>
-            <p>We have sent you a confirmation email with the cash receipt.</p>
+            <p class="">We’ve mailed you the cash receipt. Please click <b>"Get ticket"</b> to receive your ticket(s).</p>
```
Comment (jace) [inline boxoffice/templates/boxoffice.html]: The "please click" sentence is redundant and can be removed.

File: boxoffice/templates/boxoffice.html [hunk 8]
```diff
@@ -166,11 +166,10 @@ <h1 class="category-heading">{{title}}</h1>
...
             <div class="receipt-btn-wrapper">
               <a href="{{ cashReceiptURL }}" target="_blank" class="boxoffice-button boxoffice-button-action receipt-button" on-click="fire('eventAnalytics','view cash receipt', 'cash receipt')">View cash receipt</a>
-              <a href="{{ attendeeAssignmentURL }}" target="_blank" class="boxoffice-button boxoffice-button-action receipt-button" on-click="fire('eventAnalytics','fill attendee details', 'attendee details')">Fill attendee details</a>
+              <a href="{{ attendeeAssignmentURL }}" target="_blank" class="boxoffice-button boxoffice-button-action receipt-button" on-click="fire('eventAnalytics','fill attendee details', 'attendee details')">Get ticket</a>
```
Comment (jace) [inline boxoffice/templates/boxoffice.html]: Get ticket button should come first.

File: boxoffice/templates/order_confirmation_mail.html [hunk 9]
```diff
@@ -144,9 +145,9 @@
 
     <p>Hello {{ order.buyer_fullname }},</p>
 
-    <p class="content">Thanks for your interest in {{ order.item_collection.title }}. Please take a moment to tell us who will be attending on your ticket(s).</p>
+    <p class="content">Thanks for your interest in {{ order.item_collection.title }}. If you haven’t got your ticket(s) yet, please click on “Get ticket” below.</p>
```
Comment (jace) [inline boxoffice/templates/order_confirmation_mail.html]: Don't ask the user to click on a button.

File: boxoffice/templates/order_confirmation_mail.html [hunk 10]
```diff
@@ -226,8 +227,9 @@ <h1 class="header">Cash Receipt</h1>
     
     <div class="terms">
       <h5 class="terms-heading">Terms &amp; Conditions</h5>
-      <p class="terms-content">For terms of service and refund policy, please visit <a href="https://hasgeek.com/about/policy/">https://hasgeek.com/about/policy</a></p>
-      <p class="terms-content">For further queries, please write to <a href="mailto:support@hasgeek.com">support@hasgeek.com</a></p>
+      <p class="terms-content">For terms of service and refund policy, please visit <a href="https://hasgeek.com/about/policy/">https://hasgeek.com/about/policy</a>.</p>
+      <p class="terms-content">For further queries, please write to <a href="mailto:support@hasgeek.com">support@hasgeek.com</a>.</p>
+      <p class="terms-content">Print your cash receipt <a href={{"{base_url}/order/{access_token}/receipt".format(base_url=base_url, access_token=order.access_token)}} target="_blank">here</a>.</p>
```
Comment (jace) [inline boxoffice/templates/order_confirmation_mail.html]: This text should be in the org data, not hardcoded into the templates.

File: boxoffice/templates/ticket_assignment_mail.html [hunk 11]
```diff
@@ -0,0 +1,112 @@
+<!DOCTYPE html>
+<html lang="en">
+
+<head>
+  <meta charset='UTF-8'>  
+  <title>{{ org.title }}</title>  
+  <link rel="stylesheet" type="text/css" href="//fonts.googleapis.com/css?family=Source+Sans+Pro:400italic,600italic,400,600" />
```
Comment (jace) [inline boxoffice/templates/ticket_assignment_mail.html]: Don't depend on this working in email.

File: boxoffice/templates/ticket_assignment_mail.html [hunk 12]
```diff
@@ -0,0 +1,112 @@
...
+    }
+  </style>
+</head>
+
+<body>
+  <div>
+
+    <p>Hello {{ line_item.current_assignee.fullname }},</p>
+
+    <p>Congratulations! You have now got a ticket for {{line_item.item.title|replace("ticket", "")|trim}}.</p>
```
Comment (jace) [inline boxoffice/templates/ticket_assignment_mail.html]: Change "You have now got a ticket for" to "You have a ticket for"

File: boxoffice/templates/ticket_assignment_mail.html [hunk 13]
```diff
@@ -0,0 +1,112 @@
...
+        </tr>
+      </table>
+    </div>
+
+    <p>All you have to do is walk-in with your photo id card and this email.</p>
+    <p>For this order (receipt No: {{order.invoice_no}}), transfer ticket or edit your details <a href={{"{base_url}/order/{access_token}/ticket".format(base_url=base_url, access_token=order.access_token)}} target="_blank">here</a></p>
+    <p>If you want to cancel your ticket, please mail to <a href="mailto:{{org.details.get('support_email')}}">{{org.details.get('support_email')}}</a> with your receipt number.</p>
+
+    <p>Thank you,</p>
+    <p>For {{ org.title }}</p>
```
Comment (jace) [inline boxoffice/templates/ticket_assignment_mail.html]: The text of this template should be in org data too.

File: boxoffice/views/participant.py [hunk 14]
```diff
@@ -20,7 +21,7 @@ def assign(order):
     """
     assignee_dict = request.json.get('attendee')
     if not request.json or not assignee_dict or not assignee_dict.get('email') or not assignee_dict.get('fullname'):
-        return make_response(jsonify(message='Missing Attendee details'), 400)
+        return make_response(jsonify(status='error', error='missing_attendee_details', error_description='Attendee details are missing'), 400)
```
Comment (jace) [inline boxoffice/views/participant.py]: User-facing strings should be in double quotes.

**Pull request merged**